siteroot/urls.py

```diff
@@ -13,20 +13,24 @@
     1. Import the include() function: from django.urls import include, path
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
+from django.conf import settings
 from django.contrib.auth import views as auth_views
 from django.urls import path, include
+from django.conf import settings
 
 from bookmarks.admin import linkding_admin_site
 from .settings import ALLOW_REGISTRATION, DEBUG
 
 urlpatterns = [
-    path('admin/', linkding_admin_site.urls),
-    path('login/', auth_views.LoginView.as_view(redirect_authenticated_user=True,
-                                                extra_context=dict(allow_registration=ALLOW_REGISTRATION)),
-         name='login'),
-    path('logout/', auth_views.LogoutView.as_view(), name='logout'),
-    path('change-password/', auth_views.PasswordChangeView.as_view(), name='change_password'),
-    path('password-change-done/', auth_views.PasswordChangeDoneView.as_view(), name='password_change_done'),
+    path(settings.LD_CONTEXT_PATH, include([
```

Comment:
This nesting is pretty clever, and seems to work well from what I've tested so far. 👍

As an alternative to nesting the URLs both in the bookmark app, and in the siteroot app, it would be enough to just add this block to the end of the file:
```python
if settings.LD_CONTEXT_PATH:
    urlpatterns = [path(settings.LD_CONTEXT_PATH, include(urlpatterns))]
```
That would basically just take all the URLs used in all applications, and move them under the context path. I would prefer it that way, as I think it's a bit more readable and clearer how the setting is applied. But it also breaks the tests that you added, so these would have to be changed in that case. Also not a blocker, but if we can figure out an alternative for the tests, we could consider making this change.

Reply:
This works with the tests suggested above.

Reply:
It looks great, with less code and more intuitive logic. I never thought about it before.